akka-http-testkit/src/test/java/akka/http/javadsl/testkit/JUnitRouteTestTest.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2009-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.http.javadsl.testkit;
+
+import java.util.function.Function;
+import java.util.concurrent.CompletableFuture;
+
+import org.junit.Test;
+import akka.http.javadsl.testkit.*;
+
+import akka.http.javadsl.model.*;
+import akka.http.javadsl.model.headers.*;
+import akka.http.javadsl.server.Rejections;
+
+public class JUnitRouteTestTest extends JUnitRouteTest {
+
+  @Test
+  public void testTheMostSimpleAndDirectRouteTest() {
+    TestRoute route =
+      testRoute(
+        complete(HttpResponse.create())
+      );
+
+    route.run(HttpRequest.GET("/"))
+      .assertStatusCode(StatusCodes.OK);
+  }
+
+  @Test
+  public void testUsingADirectiveAndSomeChecks() {
+    RawHeader pinkHeader = RawHeader.create("Fancy", "pink");
+
+    TestRoute route =
+      testRoute(
+        respondWithHeader(pinkHeader, () ->
+          complete("abc")
+        )
+      );
+
+    route.run(HttpRequest.GET("/").addHeader(pinkHeader))
+      .assertStatusCode(StatusCodes.OK)
+      .assertContentType("text/plain; charset=UTF-8")
+      .assertEntity("abc")
+      .assertHeaderExists("Fancy", "pink");
+  }
+
+  @Test
+  public void testProperRejectionCollection() {
+    TestRoute route =
+      testRoute(
+        get(() ->
+          complete("naah")
+        ).orElse(
+          put(() ->
+            complete("naah")
+          )
+        )
+      );
+
+    route.run(HttpRequest.POST("/abc").withEntity("content"))
+      /* XXX: Expected?
+      .assertRejections(
+        Rejections.method(HttpMethods.GET),
+        Rejections.method(HttpMethods.PUT)
+      )
```

Comment:
Indeed the javadsl `RouteTest` does `runRoute` rather than `runRouteUnSealed`. It seems a lot more consistent to use `runRouteUnSealed` there, but that'd mean:
* updating a lot of tests that currently rely on tests being sealed
* providing a nice way to seal a route in the javadsl